src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1887,8 +1892,20 @@ protected override void HandleEvaluateRequestAsync(IRequestResponder<EvaluateArg
                 flags |= enum_EVALFLAGS.EVAL_NOSIDEEFFECTS;
             }
 
+            if (context == EvaluateArguments.ContextValue.Clipboard)
+            {
+                dapEvalFlags |= DAPEvalFlags.CLIPBOARD_CONTEXT;
+            }
+
             IDebugProperty2 property;
-            hr = expressionObject.EvaluateSync(flags, Constants.EvaluationTimeout, null, out property);
+            if (dapEvalFlags != DAPEvalFlags.NONE)
```

Comment:
> if (dapEvalFlags != DAPEvalFlags.NONE) [](start = 11, length = 39)

This check doesn't seem to make sense -- either you should do: `if (expressionObject is IDebugExpressionDAP expressionDapObject)` or you should just always call it. I would probably argue for the `if` check since someday we may use OpenDebugAD7 for hosting other engines.